### ISSUE TITLE
V12 Support

### ISSUE THREAD

**Issue by rmasoni** [mclemente/fvtt-condition-lab-triggler#51]
Hey, thanks for the wonderful job you've done so far since taking over the module.

Should we expect an update for V12 anytime soon? I've been struggling with the core conditions.

**Comment by mclemente** [comment]
I'm sorry, but don't expect anything new in the coming weeks.
The module still isn't 100% since dnd5e 3.0's release with the way it changed conditions and I can't guarantee it works properly on V12.

**Comment by rmasoni** [comment]
Oh, bummer! Core 5e combat is so bad without the extra conditions, and to make things worse Token Health also stopped working. It was a nightmare to run my last session.

So the main issue is with the system, not V12? I wish I could help, but I'm not proficient with JS.

Thank you anyways! I hope we get an update some time in the future.

**Comment by mclemente** [comment]
Yeah, dnd5e added so many new things that I need to fix it for the system specifically while trying to not break the module for every other system.

The main issue is that I've got really busy around the time V12 was coming out, though.

Ah, about Token Health, I've recently have been commissioned to fix it! I just haven't claimed the package on the FoundryVTT site yet. You can get it here https://github.com/mclemente/fvtt-token-health/releases/latest

**Comment by rmasoni** [comment]
That's one good news! Thanks!

**Comment by FoxLee** [comment]
> I'm sorry, but don't expect anything new in the coming weeks. The module still isn't 100% since dnd5e 3.0's release with the way it changed conditions and I can't guarantee it works properly on V12.

If you don't mind my asking, is there a lot to be done aside from what D&D5e needs? I understand why you would not do a proper v12 release without that completed, but for those of us whose systems have not changed much, I'm curious as to whether it would be worth forking in the mean time.

**Comment by mclemente** [comment]
@FoxLee I'll release a V12 version that doesn't support D&D5e, then. There wasn't much to change besides deprecation warnings and some changes to hooks.

**Comment by FoxLee** [comment]
> @FoxLee I'll release a V12 version that doesn't support D&D5e, then. There wasn't much to change besides deprecation warnings and some changes to hooks.

Oh that would be great, cheers from the 4e crowd :D

**Comment by Stexinator** [comment]
@mclemente I saw you already made a v12 version. Unfortunatley it seems that it does not support the new/renamed fields of status effects
icon => img
label => name

Should I open a new issue for this?

**Comment by mclemente** [comment]
@Stexinator you mean the opposite, right? It doesn't support icon/label and only supports the img/name keys.

I can add the getter/setter like Core does, I just thought it wasn't needed for the module. What issues are you running into?

**Comment by Stexinator** [comment]
It does not show the image and does not translate the keys on v12 with a system that already uses the new fields

**Comment by mclemente** [comment]
@Stexinator I've released a new version that should fix these issues.

**Comment by Stexinator** [comment]
![grafik](https://github.com/user-attachments/assets/d929f1c9-9efd-42f8-a66e-275c30edbc7d)

Unfortunately the new version does not work as expected. No errors on console

**Comment by Strange-Dust** [comment]
Hi, I'm not sure if there is something I am doing wrong, but basically the custom conditions don't apply when I click on them in the "Assign Status Effects" box on the token.

I can get into the menus to customize them, no problem.
New stuff shows up with all the other conditions, no problem.
But when I click the new one, nothing happens.  Default conditions still work fine.

Also, I tried using output to chat, and output default to chat too.  If I add the Dead condition, that one does go to chat, but nothing else gets outputted to chat when adding or removing conditions. 

Foundry v12.330
Pathfinder 2e 6.2.1


Thank you

**Comment by Strange-Dust** [comment]
I looked into it a bit further and it looks like it is because of how the pf2e system uses items instead of active effects, not because of a v12 compatibility thing, sorry.

Too bad, because this module is quite nice.

**Comment by Angellorye** [comment]
![image](https://github.com/user-attachments/assets/4d91ba5a-5394-4c01-98dd-2bc9d3b5846f)
That 'is not working T^T

**Comment by mclemente** [comment]
@Angellorye that issue should be fixed now. You'll need to reset the condition lab's mapping, though.

**Comment by Fridan99** [comment]
Any news about dnd5e compatibility?

Foundry v12.331
dnd5e 3.3.1

With these versions it doesn't seem to work correctly. At least in the token status hud the conditions do not appear (only the exhaust one). 🤪

**Comment by andaug** [comment]
Same as [Stexinator](https://github.com/Stexinator) above. And ootb status effects missing icon and cant be applied, on the token->right click->Assign efffects. Only the my added effect shows and can be applied even if I see the same error --No reference entry... -- in the config screen. Using it in Vtt12 for Twilight 2000 4e system with the latest year zero modules.
